config/merge_config.go

```diff
@@ -0,0 +1,393 @@
+	} else {
+		agentConf.APIEndpoints = []APIEndpoint{{APIKey: vals[0]}}
+	}
+}
+
+// Uses viper config to retrieve both yaml and env variables
+func mergeConfig(dc ddconfig.Config, agentConf *AgentConfig) error {
+	apiKey := dc.GetString("api_key")
+
+	if apiKey != "" {
+		mergeAPIKeys(apiKey, agentConf)
+	}
+
+	en := dc.GetString(keyEnabled)
+	if enabled, err := isAffirmative(en); enabled {
+		agentConf.Enabled = true
+		agentConf.EnabledChecks = processChecks
+	} else if strings.ToLower(en) == "disabled" {
+		agentConf.Enabled = false
+	} else if !enabled && err == nil {
+		agentConf.Enabled = true
+		agentConf.EnabledChecks = containerChecks
+	}
+
+	url, err := url.Parse(ddconfig.GetMainEndpointWithConfig(dc, "https://process.", "process_config.process_dd_url"))
+	if err != nil {
+		return fmt.Errorf("error parsing process_dd_url: %s", err)
+	}
+	if len(agentConf.APIEndpoints) > 0 {
+		agentConf.APIEndpoints[0].Endpoint = url
+	} else {
+		agentConf.APIEndpoints = []APIEndpoint{{Endpoint: url}}
+	}
+
+	if enabled, err := isAffirmative(dc.GetString("LOGS_STDOUT")); err == nil {
+		agentConf.LogToConsole = enabled
+	}
+	if enabled, err := isAffirmative(dc.GetString("LOG_TO_CONSOLE")); err == nil {
+		agentConf.LogToConsole = enabled
+	}
+
+	if dc.IsSet(keyLogFile) {
+		agentConf.LogFile = dc.GetString(keyLogFile)
+	}
+
+	if dc.IsSet(keyIntervalsContainer) {
+		agentConf.CheckIntervals["container"] = time.Duration(dc.GetInt(keyIntervalsContainer)) * time.Second
+	}
+	if dc.IsSet(keyIntervalsContainerRT) {
+		agentConf.CheckIntervals["rtcontainer"] = time.Duration(dc.GetInt(keyIntervalsContainerRT)) * time.Second
+	}
+	if dc.IsSet(keyIntervalsProcess) {
+		agentConf.CheckIntervals["process"] = time.Duration(dc.GetInt(keyIntervalsProcess)) * time.Second
+	}
+	if dc.IsSet(keyIntervalsProcessRT) {
+		agentConf.CheckIntervals["rtprocess"] = time.Duration(dc.GetInt(keyIntervalsProcessRT)) * time.Second
+	}
+	if dc.IsSet(keyIntervalsConnections) {
+		agentConf.CheckIntervals["connections"] = time.Duration(dc.GetInt(keyIntervalsConnections)) * time.Second
+	}
+
+	if dc.IsSet(keyBlacklistPatterns) {
+		blackPat := dc.GetStringSlice(keyBlacklistPatterns)
+		blacklist := make([]*regexp.Regexp, 0, len(blackPat))
+		for _, b := range blackPat {
+			r, err := regexp.Compile(b)
+			if err != nil {
+				log.Warnf("Invalid blacklist pattern: %s", b)
+			}
+			blacklist = append(blacklist, r)
+		}
+		agentConf.Blacklist = blacklist
+	}
+
+	if dc.IsSet(keyScrubArgs) {
+		agentConf.Scrubber.Enabled = dc.GetBool(keyScrubArgs)
+	}
+
+	if dc.IsSet(keyCustomSensitiveWords) {
+		csw := dc.GetString(keyCustomSensitiveWords)
+		agentConf.Scrubber.AddCustomSensitiveWords(strings.Split(csw, ","))
+	}
+
+	if dc.IsSet(keyStripProcessArguments) {
+		agentConf.Scrubber.StripAllArguments = dc.GetBool(keyStripProcessArguments)
+	}
+
+	if dc.IsSet(keyQueueSize) {
+		agentConf.QueueSize = dc.GetInt(keyQueueSize)
+	}
+
+	if dc.IsSet(keyMaxProcFDs) {
```

Comment:
There's a lot of duplication here, can we create a helper function to reduce a lot of it?

```go
func ifSet(dc ddconfig.Config, key string, f func(s string)) {
  if dc.IsSet(key) {
    f(s)
  }
}

... 

ifSet(dc, keyMaxProcFDs, func(key string) {
   agentConf.MaxProcFDs = dc.GetInt(key)
})
```

Reply:
I added helpers